homeassistant/components/aquacell/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for Aquacell integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aioaquacell import ApiException, AquacellApi, AuthenticationFailed
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_REFRESH_TOKEN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(
+    hass: HomeAssistant, email: str, password: str
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    session = async_get_clientsession(hass)
+    api = AquacellApi(session)
+    refresh_token = None
+    try:
+        refresh_token = await api.authenticate(email, password)
+    except AuthenticationFailed as err:
+        raise InvalidAuth from err
+    except ApiException as err:
+        raise CannotConnect from err
+
+    return {
+        CONF_REFRESH_TOKEN: refresh_token,
+        CONF_EMAIL: email,
+        CONF_PASSWORD: password,
+    }
+
+
+class AquaCellConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aquacell."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(
```

Comment:
I think we can inline this function now (and don't worry I know about the reauth, there are some tricks to reuse this flow to avoid code duplication)